packages/babel-core/src/transformation/plugin-pass.ts

```diff
@@ -1,20 +1,20 @@
 import type File from "./file/file";
 import type { NodeLocation } from "./file/file";
 
-export default class PluginPass {
+export default class PluginPass<Options = {}> {
   _map: Map<unknown, unknown> = new Map();
-  key: string | undefined | null;
-  file: File;
-  opts: any;
+  declare key: string | undefined | null;
+  declare file: File;
+  declare opts: Partial<Options>;
 
   // The working directory that Babel's programmatic options are loaded
   // relative to.
-  cwd: string;
+  declare cwd: string;
 
   // The absolute path of the file being compiled.
-  filename: string | void;
+  declare filename: string | void;
```

Comment:
We shouldn't use `declare` here, since the constructor then sets these property (and we do not want to walk the prototype checking if they are already there)

Reply:
I mark them as `declare` because currently class properties are transpiled to initialization assignments in constructor as we are targeting Node.js 6: https://unpkg.com/@babel/core@7.22.9/lib/transformation/plugin-pass.js, this change will get rid of such extra assignments.

But since we will ship native class properties in Babel 8, I am happy to revert that.